nifi-nar-bundles/nifi-aws-bundle/nifi-aws-parameter-providers/src/main/java/org/apache/nifi/parameter/aws/AwsSecretsManagerParameterProvider.java

```diff
@@ -0,0 +1,305 @@
+                final String parameterName = field.getKey();
+                final String parameterValue = field.getValue().textValue();
+                if (parameterValue == null) {
+                    logger.debug("Secret [{}] key [{}] has no value", secretName, parameterValue);
+                    continue;
+                }
+
+                parameters.add(createParameter(parameterName, parameterValue));
+            }
+
+            groups.add(new ParameterGroup(secretName, parameters));
+
+            return groups;
+        } catch (final ResourceNotFoundException e) {
+            logger.error("Secret [{}] not found", secretName);
+            throw new IllegalStateException(String.format("Secret %s not found", secretName), e);
+        } catch (final AWSSecretsManagerException e) {
+            logger.error("Error retrieving secret [{}]", secretName);
+            throw new IllegalStateException("Error retrieving secret " + secretName, e);
+        }
+    }
+
+    private Parameter createParameter(final String parameterName, final String parameterValue) {
+        final ParameterDescriptor parameterDescriptor = new ParameterDescriptor.Builder().name(parameterName).build();
+        return new Parameter(parameterDescriptor, parameterValue, null, true);
+    }
+
+    protected ClientConfiguration createConfiguration(final ConfigurationContext context) {
+        final ClientConfiguration config = new ClientConfiguration();
+        config.setMaxErrorRetry(0);
+        config.setUserAgentPrefix(DEFAULT_USER_AGENT);
+        config.setProtocol(DEFAULT_PROTOCOL);
+        final int commsTimeout = context.getProperty(TIMEOUT).asTimePeriod(TimeUnit.MILLISECONDS).intValue();
+        config.setConnectionTimeout(commsTimeout);
+        config.setSocketTimeout(commsTimeout);
+
+        if (this.getSupportedPropertyDescriptors().contains(SSL_CONTEXT_SERVICE)) {
+            final SSLContextService sslContextService = context.getProperty(SSL_CONTEXT_SERVICE).asControllerService(SSLContextService.class);
+            if (sslContextService != null) {
+                final SSLContext sslContext = sslContextService.createContext();
+                SdkTLSSocketFactory sdkTLSSocketFactory = new SdkTLSSocketFactory(sslContext, new DefaultHostnameVerifier());
+                config.getApacheHttpClientConfig().setSslSocketFactory(sdkTLSSocketFactory);
+            }
+        }
+
+        return config;
+    }
+
+    public static AllowableValue createAllowableValue(final Regions region) {
+        return new AllowableValue(region.getName(), region.getDescription(), "AWS Region Code : " + region.getName());
+    }
+
+    public static AllowableValue[] getAvailableRegions() {
+        final List<AllowableValue> values = new ArrayList<>();
+        for (final Regions region : Regions.values()) {
+            values.add(createAllowableValue(region));
+        }
+        return values.toArray(new AllowableValue[0]);
+    }
```

Comment:
Recommend moving this methods higher in the file following general conventions for public static methods being declared before less visible methods. Do they need to `public` or should they be package-private or private?